compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.h

```diff
@@ -38,14 +41,33 @@ struct GPUMMAHeuristicSeeds {
 struct GPUMMASchedule {
   // Index of the chosen intrinsic into the list of given MMA intrinsics
   uint64_t index;
-  int64_t mSize;      // Native MMA size along M dimension
-  int64_t nSize;      // Native MMA size along N dimension
-  int64_t kSize;      // Native MMA size along K dimension
-  int64_t mWarpCount; // Number of subgroups along M dimension
-  int64_t nWarpCount; // Number of subgroups along N dimension
-  int64_t mTileCount; // Number of tiles per subgroup along M dimension
-  int64_t nTileCount; // Number of tiles per subgroup along N dimension
-  int64_t kTileCount; // Number of tiles along K dimension
+  int64_t mSize; // Native MMA size along M dimension
+  int64_t nSize; // Native MMA size along N dimension
+  int64_t kSize; // Native MMA size along K dimension
+
+  // Number of subgroups along M dimensions
+  SmallVector<int64_t> mSubgroupCounts;
+  // Number of subgroups along N dimensions
+  SmallVector<int64_t> nSubgroupCounts;
+  // Number of tiles per subgroup along M dimensions
+  SmallVector<int64_t> mTileCounts;
+  // Number of tiles per subgroup along N dimensions
+  SmallVector<int64_t> nTileCounts;
+  // Number of tiles along K dimensions
+  SmallVector<int64_t> kTileCounts;
+
+  GPUMMASchedule(uint64_t i, int64_t m, int64_t n, int64_t k, int64_t mSubgroup,
```

Comment:
Since we have multiple m/n/k dimensions now, can we document that we always target innermost dimension to the intrinsic? Or if we don't?